yt/data_objects/static_output.py

```diff
@@ -288,6 +288,10 @@ def __init__(self, filename, dataset_type=None, file_style=None,
         # path stuff
         self.parameter_filename = str(filename)
         self.basename = os.path.basename(filename)
+        if (    (self.basename == "")
+                and (len(filename) > 0)
+                and (filename[-1] == "/")):
+            self.basename = os.path.basename(filename[:-1])
```

Comment:
Why is `basename` only getting set conditionally here?

Reply:
It's also getting set on the above line. `os.path.basename("my_dir/plt00040/")` is `""`, while using Unix's basename utility it would be "plt00040". Auto-complete often adds that extra "/" because `plt00040` is a directory, so it's nice to handle this case.